### PR TITLE
(Feat: M2-6788) Conditional logic: Date item

### PR DESCRIPTION
**Pull request by moiskillnadne**
### 📝 Description
🔗 [Jira Ticket M2-6788](https://mindlogger.atlassian.net/browse/M2-6788)

### CODE REVIEW

File: src/shared/utils/validate/date/isFirstDateEarlier.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * the time component. If the first date is earlier, it returns `-1`.
+ * If the dates are the same, it returns `0`.
+ *
+ * @param {Date} firstDate - The first date to compare.
+ * @param {Date} secondDate - The second date to compare.
+ * @returns {boolean} - `true` if the first date is earlier than the second date, otherwise `false`.
+ */
+export const isFirstDateEarlier = (firstDate: Date, secondDate: Date): boolean => {
+  const dateOnly1 = new Date(firstDate.getFullYear(), firstDate.getMonth(), firstDate.getDate());
+  const dateOnly2 = new Date(secondDate.getFullYear(), secondDate.getMonth(), secondDate.getDate());
```
Comment (BamMironov) [inline src/shared/utils/validate/date/isFirstDateEarlier.ts:17]: Please use [startOfDay](https://date-fns.org/v3.6.0/docs/startOfDay) instead to build `dateOnly1` and `dateOnly2`

File: src/shared/utils/validate/date/isFirstDateLater.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * the time component. If the first date is earlier, it returns `-1`.
+ * If the dates are the same, it returns `0`.
+ *
+ * @param {Date} firstDate - The first date to compare.
+ * @param {Date} secondDate - The second date to compare.
+ * @returns {boolean} - `true` if the first date is later than the second date, otherwise `false`.
+ */
+export const isFirstDateLater = (firstDate: Date, secondDate: Date): boolean => {
+  const dateOnly1 = new Date(firstDate.getFullYear(), firstDate.getMonth(), firstDate.getDate());
+  const dateOnly2 = new Date(secondDate.getFullYear(), secondDate.getMonth(), secondDate.getDate());
```
Comment (BamMironov) [inline src/shared/utils/validate/date/isFirstDateLater.ts:17]: Please use [startOfDay](https://date-fns.org/v3.6.0/docs/startOfDay) instead to build `dateOnly1` and `dateOnly2`

**Review by BamMironov — CHANGES_REQUESTED** [review]
Please consider little changes below

File: src/shared/utils/validate/date/isFirstDateEarlier.ts (hunk 1 above)
Comment (moiskillnadne) [reply src/shared/utils/validate/date/isFirstDateEarlier.ts]: Done

File: src/shared/utils/validate/date/isFirstDateLater.ts (hunk 2 above)
Comment (moiskillnadne) [reply src/shared/utils/validate/date/isFirstDateLater.ts]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**